ccm_web/server/src/api/dtos/api.section.external.users.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ArrayMaxSize,
+  ArrayMinSize,
+  IsArray,
+  IsNotEmpty,
+  ValidateNested
+} from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+import { Type } from 'class-transformer'
+import { UserEnrollmentType } from '../../canvas/canvas.interfaces'
+
+export class SectionExternalUserDto {
+  @ApiProperty()
+  @IsNotEmpty()
+  email: string
+
+  @ApiProperty()
+  @IsNotEmpty()
+  givenName: string
+
+  @ApiProperty()
+  @IsNotEmpty()
+  surname: string
+
+  @ApiProperty({ enum: UserEnrollmentType })
+  @IsNotEmpty()
+  type: UserEnrollmentType
+
+  constructor (email: string, givenName: string, surname: string, type: UserEnrollmentType) {
+    this.email = email
+    this.givenName = givenName
+    this.surname = surname
+    this.type = type
+  }
+}
+
+export class SectionExternalUsersDto {
+  @ApiProperty({ type: [SectionExternalUserDto] })
+  @IsArray()
+  @ArrayMinSize(1)
+  @ArrayMaxSize(400)
```

Comment:
is 400 going to be the limit of users we will allow through CSV file?  as we did with UM user.

Reply:
I think we said `200`, because of the extra calls involved.

Reply:
I can reduce that, then.

Reply:
🏁